calico/openstack/mech_calico.py

```diff
@@ -100,26 +103,52 @@ def __init__(self):
         # initialize these properly when we first need them.
         self.db = None
         self.transport = None
-        self._state_initialized = False
+        self._my_pid = None
+        self._periodic_resync_greenlet = None
+        self._epoch = 0
 
-        # Start our resynchronization process.
-        eventlet.spawn(self.periodic_resync_thread)
+        # Make sure we initialise even if we don't see any API calls.
+        eventlet.spawn_after(STARTUP_DELAY_SECS, self._init_state)
 
     def _init_state(self):
         """
         Creates the connection state required for talking to the Neutron DB
         and to etcd. This is a no-op if it has been executed before.
         """
-        if self._state_initialized:
+        current_pid = os.getpid()
+        if self._my_pid == current_pid:
+            # We've initialised our PID and it hasn't changed since last time,
+            # nothing to do.
```

Comment:
Do we really want to do this on *every* call? It's not clear that we do if we can possibly avoid it.